azure/Guidelines.md

```diff
@@ -18,6 +20,35 @@ Teams building ARM Resource Providers (RPs) MUST follow the additional guidance
 
 ARM RPs are Azure Fundamentals requirement for Azure Services and ARM RP review is another mandatory review. Some of the guidance overlaps with general API review, but passing one review will generally make the other one go very quickly.
 
+### Advice for new services
+
+Developing a new service requires the development of at least 1 (management plane) API and potentially one or more additional (data plane) APIs.  
+
+* Think about naming from the context of a **developer experience**.  
+  * Start with the "things" your API manipulates, then think about the operations that a developer needs to do to these "things".
+  * Avoid the use of generic names like "Object", "Job", "Task", "Operation" (for example - the list is not exhaustive).
+  * What happens to the names when the focus of the service expands?  It may be worth starting with a less generic name to avoid a breaking change later on.
+* Use previews to get the shape right.  There are different notification, breaking change, and lifetime requirements on preview API versions.
+  * We recommend a minimum of 2 preview versions prior to your first GA release.  
+  * Your API should be in preview for a minimum of 3 months to gain customer insights on usage.
+
+Breaking changes are a source of concern during initial review.  Without a history, the review process attempts to identify patterns that may result in breaking changes later on.  For instance:
+
+* Return collections with server-side paging, even if your resource does not currently need paging.  This avoids a breaking change when your service expands.
+* Think about how the developer can reason about the collection organization.  Filtering is a common customer request.
+* Prefer allowing the developer to specify the ID of the resource.  A developer will commonly want to download a specific resource by name.
+* Use extensible enumerations unless you are completely sure that the enumeration will never expand.  Extensible enumerations are modeled as strings - expanding an extensible enumeration is not a breaking change.
```

Comment:
Is it necessary to mentioned that even with a extensible enumeration / string there is a small likelihood that a client might not be able to deal with new values ?

Reply:
I think having a blurb about forward-compat and clients would be great to include because I've seen a lot of folks surprised by it.

Reply:
Yes this is true. I'd to use the term "enumerations" for a forever set of fixed values. And never use the term "enumerations" for non-fixed values; just use strings.